shaq/_cli.py

```diff
@@ -164,11 +164,74 @@ def _parser() -> argparse.ArgumentParser:
         type=str,
         help="send the request to a proxy server",
     )
+    advanced_group.add_argument(
+        "--metadata",
+        action="store_true",
+        help="Overwrite titel and metadata of the song",
+    )
     return parser
 
+def updateMetadata(console, title, artist, path):
+    
+    # Imports happen here so that when this functionality is not used, mutagen is not needed to run the program
+    # This is to avoid unnecessary dependencies.
+
+    try:
+        from mutagen.flac import FLAC
+        from mutagen.mp3 import MP3
+        from mutagen.id3 import TPE2, TIT2, TPE1
+        from mutagen.oggvorbis import OggVorbis
+    except ImportError:
+        console.print("[red]mutagen library is not installed. Install it with 'pip install mutagen'[/red]")
+        sys.exit(1)
+
+    ext = path.suffix.lower()
+    try:
+        audio = None
+        tag_map = {      
+            ".flac":  (FLAC, {"title": str, "artist": str, "albumartist": str}),
+            ".ogg":   (OggVorbis, {"title": str, "artist": str, "albumartist": str})
+
+        }
+
+        # mp3 is separate because it uses ID3 tags
+        if ext == ".mp3":
+            audio = MP3(path)
+            if audio.tags is None:
+                audio.add_tags()
+            audio["TIT2"] = TIT2(encoding=3, text=title) # Title
+            audio["TPE1"] = TPE1(encoding=3, text=artist) # Contributing artist
+            audio["TPE2"] = TPE2(encoding=3, text=artist)  # Album artist
+            audio.save()
+        elif ext in tag_map:
+            cls, keys = tag_map[ext]
+            audio = cls(path)
+            for key in keys:
+                if key == "title":
+                    audio[key] = title
+                elif key == "artist":
+                    audio[key] = artist
+                elif key == "albumartist":
+                    audio[key] = artist
+            audio.save()
+        else:
+            console.print(f"[yellow]File type {ext} not supported for metadata writing.[/yellow]")
+            return
```

Comment:
Nitpick: This kind of "late return" is a bit harder to follow than an "early return" pattern. Ideally this would be at the top of the function, i.e.:

```
if ext not in tag_map:
    # log
    return
```

...that way the rest of the code can stay at a smaller indentation :slightly_smiling_face: